app/dashboard/browse/search/organizations/[id]/history/donorHistory.tsx

```diff
@@ -0,0 +1,324 @@
+      let sortedArray = grantList;
+      sortedArray.sort((a, b) =>
+        a.name > b.name ? 1 : b.name > a.name ? -1 : 0
+      );
+      setGrantList(sortedArray);
+    } else {
+      let sortedArray = grantList;
+      sortedArray.sort((a, b) =>
+        b.name > a.name ? 1 : a.name > b.name ? -1 : 0
+      );
+      setGrantList(sortedArray);
+    }
+    setNameLeastToGreatest(!nameLeastToGreatest);
+  };
+
+  const handleAmountSort = () => {
+    // Resetting other filters
+    setAmountFilter(true);
+    setNameFilter(false);
+    setNameLeastToGreatest(false);
+    setDateFilter(false);
+    setDateLeastToGreatest(false);
+    // Sort
+    if (amountLeastToGreatest) {
+      let sortedArray = grantList;
+      sortedArray.sort((a, b) => a.amount - b.amount);
+      setGrantList(sortedArray);
+    } else {
+      let sortedArray = grantList;
+      sortedArray.sort((a, b) => b.amount - a.amount);
+      setGrantList(sortedArray);
+    }
+    setAmountLeastToGreatest(!amountLeastToGreatest);
+  };
+
+  const handleDateSort = () => {
+    // Resetting other filters
+    setDateFilter(true);
+    setNameFilter(false);
+    setNameLeastToGreatest(false);
+    setAmountFilter(false);
+    setAmountLeastToGreatest(false);
+    // Sort
+    if (dateLeastToGreatest) {
+      let sortedArray = grantList;
+      sortedArray.sort((a, b) =>
+        new Date(a.created_at) > new Date(b.created_at)
+          ? 1
+          : new Date(b.created_at) > new Date(a.created_at)
+          ? -1
+          : 0
+      );
+      setGrantList(sortedArray);
+    } else {
+      let sortedArray = grantList;
+      sortedArray.sort((a, b) =>
+        new Date(b.created_at) > new Date(a.created_at)
+          ? 1
+          : new Date(a.created_at) > new Date(b.created_at)
+          ? -1
+          : 0
+      );
+      setGrantList(sortedArray);
+    }
+    setDateLeastToGreatest(!dateLeastToGreatest);
+  };
+
+  // Row component for table
+  function TableRow({
+    id,
+    name,
+    amount,
+    orgFunded,
+    date,
+  }: {
+    id: number;
+    name: string;
+    amount: number;
+    orgFunded: { name: string } | null;
+    date: string;
+  }) {
+    const dateString =
+      date.slice(5, 7) + "/" + date.slice(8, 10) + "/" + date.slice(0, 4);
+    return (
+      <Link
+      href={
+        orgFunded?.name === userOrg.name
+          ? `/dashboard/organization/history/acceptedGrant/${id}`
+          : `/dashboard/organization/history/closedGrant/${id}`
+      }
+        className="flex py-4 px-6 even:bg-white odd:bg-gray-100 hover:bg-gray-200 2xl:text-base text-sm"
+      >
+        <div className="basis-1/3 truncate">{name}</div>
+        <div className="basis-1/6">${amount}</div>
+        <div className="basis-1/3 truncate">
+          {orgFunded === null ? (
+            <BlockIcon fontSize="inherit" />
+          ) : (
+            orgFunded.name
+          )}
+        </div>
+        <div className="basis-1/6">{dateString}</div>
+      </Link>
```

Comment:
Get rid of the link all together and just make it a div. This page is just for public history viewing so there isn't any need to direct users anywhere. They can do so under their own private history page

Reply:
This should also remove the need to fetch user org data